pkg/alertmanager/legacyalertmanager/provider.go

```diff
@@ -125,17 +138,24 @@ func (provider *provider) putAlerts(ctx context.Context, orgID string, alerts al
 		return err
 	}
 
-	legacyAlerts := make([]postableAlert, len(alerts))
-	for i, alert := range alerts {
-		receivers, err := cfg.ReceiverNamesFromRuleID(alert.Alert.Labels["ruleID"])
-		if err != nil {
-			return err
+	var legacyAlerts []postableAlert
+	for _, alert := range alerts {
+		ruleID, ok := alert.Alert.Labels[alertmanagertypes.RuleIDMatcherName]
+		if !ok {
+			provider.settings.Logger().WarnContext(ctx, "cannot find ruleID for alert, skipping sending alert to alertmanager", "alert", alert)
+			continue
+		}
+
+		receivers := cfg.ReceiverNamesFromRuleID(ruleID)
+		if len(receivers) == 0 {
+			provider.settings.Logger().WarnContext(ctx, "cannot find receivers for alert, skipping sending alert to alertmanager", "ruleID", ruleID, "alert", alert)
+			continue
```

Comment:
Continuing above empty channels list for all channels context, the `receivers` would be an empty list? and this change skips instead of sending notification to all channels?